src/lib/operators/join_sort_merge.cpp

```diff
@@ -126,8 +126,8 @@ class JoinSortMerge::JoinSortMergeImpl : public AbstractReadOnlyOperatorImpl {
   OperatorPerformanceData<JoinSortMerge::OperatorSteps>& _performance;
 
   // Contains the materialized sorted input tables
-  std::unique_ptr<MaterializedSegmentList<T>> _sorted_left_table;
```

Comment:
Where is it fixed in the code?

Reply:
@ArneMayer : don't know how to link to this line. Look for the removal of this line in the column materializer `auto null_rows = std::make_unique<RowIDPosList>();`.

Reply:
We don't test it and I am not sure how to test it. We would need to have a sufficiently large SMJ **and** it I don't know the chances of crashing. Might not be easy to trigger in a test (it's easy to trigger it manually with huge vector and many parallel jobs).

Reply:
Hm, that does not sound too reassuring.

Reply:
I know, but that's unfortunately true for many operators. The sanitizers did not catch it and testing it can be quite hard.

Not sure how to deal with it.

Reply:
We have a few other tests that check for concurrency issues, e.g., the [StressTests](https://github.com/hyrise/hyrise/blob/master/src/test/lib/concurrency/stress_test.cpp).

Couldn't we build a test similar to the scenario that you used when you first encountered the issue? How large is sufficiently large? If we don't know the chances of crashing, would it be an option to repeat a sufficiently large SMJ a hundred times?

Reply:
I tested it with sufficiently large data and it did not crash. The problem is the same for several parts of the system that react to input sizes (that includes the hash and sort-merge join, I think also the aggregate). They COULD potentially have problems, but they might not be triggered with data sizes that make sense in the CI.

Is this is a rather simply issue, I would argue that testing it once is fine given that is cannot be tested in the CI. That's different to the stress test that checks rather complex concurrent processes.